mango-python/bdgenomics/mango/variantViz.py

```diff
@@ -0,0 +1,442 @@
+          bins = [.00001,.0001,.01,.025,.05,.075,.1,.15,.2,.25,.3,.35,.4,.45,.5]
+          #plt.hist(self.data_bins, weights=self.data_weights, bins=500, normed=1, histtype = 'step', cumulative=1, label=self.annot_list)
+          plt.hist(self.data_bins, weights=self.data_weights, bins=bins, normed=1, histtype = 'step', label=self.annot_list)
+          plt.legend(loc='lower center')
+          #plt.yscale('log')
+          plt.xscale('log')
+          plt.xlabel("Allele Frequency")
+          plt.ylabel("Proportion of Variants (normalized)")
+          plt.show()
+        return ( sorted(self.data_bins[0])[0:20], sorted(self.data_weights[0][0:20]) )
+
+
+class VariantDistribByPop(object):
+    """
+    Plot Allele Frequency Distribution by Population
+    """
+    def __init__(self, ss, variantRDD, pop_list=[ 'AF_NFE', 'AF_AFR']):
+        """
+        :param ss: SparkContext 
+        :param variantRDD: bdgenomics VariantRDD
+        :param pop_list: list of populations
+        """
+
+        def jp3(dd1):
+            effectList = [ effect for transcript in dd1['transcriptEffects'] for effect in transcript['effects'] ]
+            mostSigEffect = most_sig_effect(effectList)
+            return mostSigEffect
+
+        data_bins = []
+        data_weights = []
+
+        def getmaf(x):
+            if(x<0.5):
+                return x
+            else:
+              return 1 - x
+
+        for pop in pop_list:
+           curr = variantRDD.toDF().rdd.map(lambda w: w.asDict()).map(lambda p: p['annotation']).filter(
+             lambda x: x['attributes'][pop] != ".").map(
+             lambda d: (jp3(d), math.floor( getmaf(float(d['attributes'][pop])) *100000)/100000)).filter(lambda f: f[1] > 0).countByValue()
+           currlist = curr.items()
+           currbins = [ item[0][1] for item in currlist ]
+           currweights = [ item[1] for item in currlist ]
+           data_bins.append(currbins)
+           data_weights.append(currweights)
+
+        self.data_bins = data_bins
+        self.data_weights = data_weights
+        self.pop_list=pop_list
+        self.sc = ss.sparkContext
+
+    def plot(self):
+        plt.rcdefaults()
+        plt.title("Allele Frequency Distribution by Population")
+        plt.hist(self.data_bins, weights=self.data_weights, bins=[.00001,.0001,.001,.01,.025,.05,.075,.1,.15,.2,.25,.3,.35,.4,.45,.5], histtype = 'step', label=self.pop_list)
+        plt.legend()
+        plt.yscale('log')
+        plt.xlabel("Minor Allele Frequency")
+        plt.ylabel("Variant Count")
+        #plt.xscale('log')
+        plt.show()
+
+
+class VariantEffectCounts(object):
+    """
+    Plot Barchart of Variant Effect Type Counts
+    """
+    def __init__(self, ss, variantRDD, includeList = []):
+        if not includeList:
+            includeList = [
+                u'non_coding_transcript_exon_variant',
+                u'3_prime_UTR_variant',
```

Comment:
This seems to be redundant to the list above

Reply:
Its not the same list, this is a default list that will give a meaningful view of many gene related variant effect counts without without being dominated by intergenic and intronic types.  The user can overide this parameter to get other categories if they choose.